src/undo/durable_move.py

```
@@ -0,0 +1,388 @@
+    if not entries:
+        return False
+    # Collapse to the latest state per (src, dst) — an entry may
+    # have been updated across crash-recovery attempts.
+    latest: dict[tuple[str, str], _JournalEntry] = {}
+    for entry in entries:
+        latest[(entry.src, entry.dst)] = entry
+    path_str = str(path)
+    for entry in latest.values():
+        if entry.state == STATE_DONE:
+            continue
+        if entry.src == path_str or entry.dst == path_str:
```

Ruff (0.15.11)

[warning] Consider merging multiple comparisons: `path_str in (entry.src, entry.dst)`. Use a `set` if the elements are hashable.

Merge multiple comparisons

(PLR1714)